File: algorithms/0435.non_overlapping_intervals/non_overlapping_intervals.rs
impl Solution {
    pub fn erase_overlap_intervals(intervals: Vec<Vec<i32>>) -> i32 {
        if intervals.is_empty() {
            return 0;
        }
        let mut counter = 0;
        let mut intervals = intervals;
        intervals.sort_by(|a, b| a[1].cmp(&b[1]));
        println!("{:?}", intervals);
        let mut prev = intervals[0][1];
        for i in 1..intervals.len() {
            if prev > intervals[i][0] {
                counter += 1;
            } else {
                prev = intervals[i][1];
            }
        }
        counter
    }
}

File: algorithms/0435.non_overlapping_intervals/non_overlapping_intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](auto &a, auto &b) {
            return a[1] < b[1];
        });
        int count = 0;
        int curr = intervals[0][1];
        for (int i = 1; i < intervals.size(); ++i) {
            const auto &interval = intervals[i];
            if (curr > interval[0]) {
                ++count;
            } else {
                curr = interval[1];
            }
        }
        return count;
    }
};